test/acceptance/helpers/helpers.go

```diff
@@ -58,6 +63,55 @@ func WaitForAllPodsToBeReady(t *testing.T, client kubernetes.Interface, namespac
 	})
 }
 
+// Deploy creates a Kubernetes deployment by applying configuration stored at filepath,
+// sets up a cleanup function and waits for the deployment to become available.
+func Deploy(t *testing.T, options *k8s.KubectlOptions, noCleanupOnFailure bool, filepath string) {
+	t.Helper()
+
+	KubectlApply(t, options, filepath)
+
+	file, err := os.Open(filepath)
+	require.NoError(t, err)
+
+	deployment := v1.Deployment{}
+	err = yaml.NewYAMLOrJSONDecoder(file, 1024).Decode(&deployment)
+	require.NoError(t, err)
+
+	Cleanup(t, noCleanupOnFailure, func() {
+		// Note: this delete command won't wait for pods to be fully terminated.
+		// This shouldn't cause any test pollution because the underlying
+		// objects are deployments, and so when other tests create these
+		// they should have different pod names.
+		KubectlDelete(t, options, filepath)
+	})
+
+	RunKubectl(t, options, "wait", "--for=condition=available", fmt.Sprintf("deploy/%s", deployment.Name))
+}
+
+// checkConnection execs into a pod of the deployment given by deploymentName
+// and runs a curl command with the provided curlArgs.
+// If expectSuccess is true, it will expect connection to succeed,
+// otherwise it will expect failure due to intentions.
+func CheckConnection(t *testing.T, options *k8s.KubectlOptions, deploymentName string, expectSuccess bool, curlArgs ...string) {
```

Comment:
Can we mention here that we're asserting on "hello world" output?

Reply:
Done!